datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -1349,18 +1349,26 @@ private void handleLeaderDoAssignment(boolean cleanUpOrphanNodes) {
         failedStreams = unclaimedTokens.keySet();
         Set<String> hosts = unclaimedTokens.values().stream().flatMap(List::stream).
             map(AssignmentToken::getIssuedFor).collect(Collectors.toSet());
-        _log.error("Stop failed to propagate within {}ms for streams: {}. The following hosts failed to claim their token(s): {}",
-            _config.getStopPropagationTimeout(), failedStreams, hosts);
+
+        if (!isNewlyElectedLeader) {
```

Comment:
This `if-else` block is unclear to me. Can you please add the comments explaining the logic?

Reply:
I already documented this and we went over this. We're skipping emitting metrics in case of leader failover because the new leader doesn't have a way to know if streams that are pending to stop have claimed their tokens. So the new leader issues new tokens and revokes them. Emitting metric in this case can cause false alerts.

Added inline comments explaining this.